File: C++/Goat Latin.cpp
// https://leetcode.com/problems/goat-latin/

class Solution {
public:
    string toGoatLatin(string S) {
        map <char, int> mp;
        mp['a'] = 1; mp['A'] = 1;
        mp['e'] = 1; mp['E'] = 1;
        mp['i'] = 1; mp['I'] = 1;
        mp['o'] = 1; mp['O'] = 1;
        mp['u'] = 1; mp['U'] = 1;
        vector<string> vect;
        vect.push_back("");
        for (auto x : S) {
            if (x == ' ')vect.push_back("");
            else vect.back() += x;
        }
        for (int i = 0; i < vect.size(); i++) {
            if (mp[vect[i][0]]) {
                vect[i] += "ma";
            }
            else {
                reverse(vect[i].begin(), vect[i].end());
                reverse(vect[i].begin(), vect[i].end() - 1);
                vect[i] += "ma";
            }
            vect[i] += string(i + 1, 'a');
        }
        string ans = "";
        for (auto x : vect)ans += x + " ";
        ans.pop_back();
        return ans;
    }
};

File: Python/Goat Latin.py
# https://leetcode.com/problems/goat-latin/

class Solution:
    def toGoatLatin(self, S: str) -> str:
        my_list  = S.split()
        vowels = ['a','A','e','E','i','I','o','O','u','U'];
        for i in range(len(my_list)):
            if my_list[i][0] in vowels:
                my_list[i] += "ma"
            else:
                my_list[i] = my_list[i][1:] + my_list[i][0] + "ma"
            my_list[i] += "a"*(i+1)
        return " ".join(my_list)
